### PR TITLE
Change ZeroMQ Socket to class

### PR DESCRIPTION
**Pull request by jez9999**
Addresses issue #21200.

### CODE REVIEW

**Comment by RyanCavanaugh** [comment]
@jez9999 Please fix the failures indicated in the Travis CI log.

**Comment by jez9999** [comment]
@RyanCavanaugh Fixed now.

**Review by erikmav — APPROVED** [review]

**Comment by RyanCavanaugh** [comment]
Approved by a listed owner. PR ready to merge pending express review by a maintainer.

**Pull request merged**
